### PR TITLE
Support for image labels

### PR DESCRIPTION
**Pull request by johnflavin**
Added methods for
* Reading Labels from Image objects
* Filtering image searches by Label:
    * Key only
    * Key=Value
    * Map of Key, Value pairs

Includes one test which creates images with labels and searches for them.

(This code is based almost entirely on the code for container labels. See PRs #285 and #313.)

### CODE REVIEW

File: src/main/java/com/spotify/docker/client/messages/ImageInfo.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/messages/ImageInfo.java]: Unused import

**Comment by davidxia** [comment]
Thanks for your PR. Things look great other than the one unused import.

**Comment by johnflavin** [comment]
Fixed. Happy to contribute.

**Comment by davidxia** [comment]
@johnflavin Great. Would you mind squashing into 1 commit? Then I'll merge.

**Comment by johnflavin** [comment]
I've never squashed commits before, but I think I made it work.

**Pull request merged**